expr/functions/moving/function.go

```diff
@@ -90,38 +90,35 @@ func (f *moving) Do(ctx context.Context, e parser.Expr, from, until int64, value
 	for _, a := range arg {
 		r := *a
 		r.Name = fmt.Sprintf("%s(%s,%s)", e.Target(), a.Name, argstr)
+
+		if windowSize == 0 {
+			result = append(result, &r)
```

Comment:
I would suggest to have a config flag here. That either can be a per-function one (e.x. `return_nans_if_step_mismatch`, by default set to True and that will populate results with NaNs) or a global one with some similar name.

Otherwise that change, while making results more expected, would break compatibility with graphite-web.

Also it might be a good idea to open a bug in graphite-project/graphite-web just for them to also consider changing the behavior of the functions.